cpp/src/utils/CMakeLists.txt

```diff
@@ -12,11 +12,7 @@ list(APPEND TS_UTILS_SOURCE_FILES ${TS_UTILS_SRC_DIR}/metrics/registry.cc)
 add_library(ts_utils SHARED ${TS_UTILS_SOURCE_FILES})
 target_include_directories(ts_utils PUBLIC ${TS_UTILS_SRC_DIR})
 target_include_directories(ts_utils PRIVATE ${Boost_INCLUDE_DIRS})
-if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
-  target_link_libraries(ts_utils ${FOLLY_LIBRARIES} ${CMAKE_DL_LIBS} ${Boost_LIBRARIES} yaml-cpp::yaml-cpp)
-else()
-  target_link_libraries(ts_utils ${FOLLY_LIBRARIES} ${CMAKE_DL_LIBS} ${Boost_LIBRARIES} yaml-cpp)
-endif()
+target_link_libraries(ts_utils ${FOLLY_LIBRARIES} ${CMAKE_DL_LIBS} ${Boost_LIBRARIES} yaml-cpp::yaml-cpp)
```

Comment:
please keep the original, otherwise mac compile will fail

Reply:
I have retained the version that is present for MAC, i.e
```
target_link_libraries(ts_utils ${FOLLY_LIBRARIES} ${CMAKE_DL_LIBS} ${Boost_LIBRARIES} yaml-cpp::yaml-cpp)
```

With
```
target_link_libraries(ts_utils ${FOLLY_LIBRARIES} ${CMAKE_DL_LIBS} ${Boost_LIBRARIES} yaml-cpp)
```
which will be used for linux inside the container, build fails with:
```
#20 702.8 [100%] Linking CXX executable torchserve_cpp_test
#20 703.5 /serve/cpp/_build/test/torchserve_cpp_test: error while loading shared libraries: libyaml-cpp.so.0.8: cannot open shared object file: No such file or directory
#20 703.5 CMake Error at /usr/share/cmake-3.28/Modules/GoogleTestAddTests.cmake:112 (message):
#20 703.5   Error running test executable.
```

Reply:
ditto